### PR TITLE
[test]  add unit test with accountType

### PR DESCRIPTION
**Pull request by huof6829**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes  https://github.com/iotexproject/iotex-core/issues/3449

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: state/account_test.go [hunk 1]
```diff
@@ -41,35 +41,69 @@ func TestNonce(t *testing.T) {
...
-	require.Equal(hash.ZeroHash256, s2.Root)
-	require.Equal(s1.CodeHash, s2.CodeHash)
+		s2 := Account{}
+		require.NoError(s2.Deserialize(ss))
+		require.Equal(s1.accountType, s2.accountType)
+		require.Equal(s1.Balance, s2.Balance)
+		require.Equal(s1.nonce, s2.nonce)
+		require.Equal(hash.ZeroHash256, s2.Root)
+		require.Equal(s1.CodeHash, s2.CodeHash)
+	})
```
Comment (dustinxie) [inline state/account_test.go:83]: lots of duplicate code, can combine into 
```
for _, test := range []struct{
    accountType 
} {
   xxxx
}
```

File: state/account_test.go [hunk 2]
```diff
@@ -41,35 +41,69 @@ func TestNonce(t *testing.T) {
...
+		require.NoError(err)
+		require.Equal(raw, hex.EncodeToString(bytes))
+	})
+	t.Run("zero nonce account type", func(t *testing.T) {
+		acct, err := NewAccount()
+		require.NoError(err)
+		require.NoError(acct.Deserialize(ss))
+		bytes, err := acct.Serialize()
+		require.NoError(err)
+		require.Equal(raw, hex.EncodeToString(bytes))
```
Comment (dustinxie) [inline state/account_test.go:105]: zero nonce type account should generate a diff raw bytes, use that to de-serialize the struct and verify it is zero nonce type

File: state/account_test.go (hunk 1 above)
Comment (huof6829) [reply state/account_test.go]: ok

File: state/account_test.go (hunk 2 above)
Comment (huof6829) [reply state/account_test.go]: Yes. done, pls review, thx

File: action/protocol/execution/protocol_test.go [hunk 3]
```diff
@@ -403,6 +404,9 @@ func (sct *SmartContractTest) prepareBlockchain(
 	if sct.InitGenesis.IsLondon {
 		cfg.Genesis.Blockchain.ToBeEnabledBlockHeight = 0
 	}
+	if sct.IsLegacyNonceAccount {
+		cfg.Genesis.Blockchain.ToBeEnabledBlockHeight = 4
+	}
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go:409]: let's use `sct.InitGenesis.IsLondon` and delete `sct.InitGenesis.IsLondon`

File: action/protocol/execution/protocol_test.go [hunk 4]
```diff
@@ -1037,6 +1041,12 @@ func TestIstanbulEVM(t *testing.T) {
 }
 
 func TestLondonEVM(t *testing.T) {
+	t.Run("factory", func(t *testing.T) {
+		NewSmartContractTest(t, "testdata-london/factory.json")
+	})
+	t.Run("factory-legacyNonceAccount", func(t *testing.T) {
+		NewSmartContractTest(t, "testdata-london/factory-legacyNonceAccount.json")
+	})
```
Comment (CoderZhi) [inline action/protocol/execution/protocol_test.go:1049]: in `London`, we should not create new legacy nonce account

File: action/protocol/execution/protocol_test.go (hunk 3 above)
Comment (huof6829) [reply action/protocol/execution/protocol_test.go]: delete sct.IsLegacyNonceAccount

**Review by CoderZhi — APPROVED** [review]

**Review by Liuhaai — APPROVED** [review]

File: state/account_test.go [hunk 5]
```diff
@@ -41,35 +40,64 @@ func TestNonce(t *testing.T) {
...
 
-	s2 := Account{}
-	require.NoError(s2.Deserialize(ss))
-	require.Equal(s1.accountType, s2.accountType)
-	require.Equal(s1.Balance, s2.Balance)
-	require.Equal(s1.nonce, s2.nonce)
-	require.Equal(hash.ZeroHash256, s2.Root)
-	require.Equal(s1.CodeHash, s2.CodeHash)
+	for _, test := range []struct {
+		acc         Account
```
Comment (dustinxie) [inline state/account_test.go:45]: `accountType: int32`

File: state/account_test.go [hunk 6]
```diff
@@ -41,35 +40,64 @@ func TestNonce(t *testing.T) {
...
+			expectedLen: 66,
+		},
+		{
+			acc: Account{
+				accountType: 0,
+			},
+			expectedLen: 64,
+		},
+	} {
+		test.acc.nonce = 0x10
```
Comment (dustinxie) [inline state/account_test.go:61]:
```
acc := Account {
    accountType: test.accountType,
    Balance: big.NewInt(20000000),
    ...
}
```

File: state/account_test.go [hunk 7]
```diff
@@ -41,35 +40,64 @@ func TestNonce(t *testing.T) {
...
-	raw := "1201301a200000000000000000000000000000000000000000000000000000000000000000"
-	ss, _ := hex.DecodeString(raw)
-	s1 := Account{}
-	require.NoError(Deserialize(&s1, ss))
-	d, err := Serialize(s1)
-	require.NoError(err)
-	require.Equal(raw, hex.EncodeToString(d))
+
+	for _, test := range []struct {
+		acc Account
```
Comment (dustinxie) [inline state/account_test.go:83]: same comment, `accountType: int32` is enough

File: action/protocol/account/protocol_test.go [hunk 8]
```diff
@@ -57,17 +57,37 @@ func TestLoadOrCreateAccountState(t *testing.T) {
...
+		require.Equal(int32(1), s.AccountType())
+		require.Equal(uint64(0x0), s.PendingNonce())
+
+		require.NoError(createAccount(sm, addrv1.String(), big.NewInt(5)))
+		s, err = accountutil.LoadAccount(sm, addrv1)
+		require.NoError(err)
+		require.Equal("5", s.Balance.String())
+		require.Equal(int32(1), s.AccountType())
+		require.Equal(uint64(0x0), s.PendingNonce())
+	})
```
Comment (dustinxie) [inline action/protocol/account/protocol_test.go:90]: too many duplicate/identical test code, use `for` loop to combine these 2 tests, same as that in  `account_test.go`

File: state/account_test.go (hunk 6 above)
Comment (huof6829) [reply state/account_test.go]: fixed

File: state/account_test.go (hunk 7 above)
Comment (huof6829) [reply state/account_test.go]: fixed

File: action/protocol/account/protocol_test.go (hunk 8 above)
Comment (huof6829) [reply action/protocol/account/protocol_test.go]: fixed

**Pull request merged**